lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -1745,6 +1745,129 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
             binder.op, resultType, data, dimValueList);
         return success();
       });
+  patterns.onOp(
+      "EyeLike", 9, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value operand, dtypeVal;
+        int64_t dtypeIntOnnx, dtypeIntTorch, diagonalIndex;
+        if (binder.tensorOperand(operand) ||
+            binder.s64IntegerAttr(dtypeIntOnnx, "dtype", -1) ||
+            binder.s64IntegerAttr(diagonalIndex, "k", 0) ||
+            binder.tensorResultType(resultType))
+          return failure();
+
+        // TODO: handle dynamic shape
+        TensorType tensorType =
+            dyn_cast<Torch::ValueTensorType>(operand.getType())
+                .toBuiltinTensor();
+        for (auto i : tensorType.getShape()) {
+          if (i == ShapedType::kDynamic)
+            return failure();
+        }
+
+        auto shapeSizes =
+            dyn_cast<Torch::ValueTensorType>(operand.getType()).getSizes();
+        Value noneVal = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+        if (dtypeIntOnnx == -1) {
+          dtypeVal = noneVal;
+        } else {
+          dtypeIntTorch = onnxDtypeIntToTorchDtypeInt(dtypeIntOnnx);
+          dtypeVal = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(dtypeIntTorch));
+        }
+
+        if (diagonalIndex == 0) {
+          // diagonalIndex = 0 populates the main diagonal
+          Value nVal = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(shapeSizes[0]));
+          if (shapeSizes[0] == shapeSizes[1]) {
```

Comment:
To clarify, if you are trying to save writing the extra value m into the IR right now, it will get written later on anyway so all that happens is the code is slightly harder to read here.